File: Java/app/src/main/java/com/javapaytaps/DetailActivity.java
package com.javapaytaps;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.LinearLayoutManager;


import com.javapaytaps.adpter.DetailAdapter;
import com.javapaytaps.databinding.ActivityDetailBinding;
import com.javapaytaps.model.DetailModel;
import com.javapaytaps.model.DataModel;

import java.util.ArrayList;

public class DetailActivity extends AppCompatActivity {

    ActivityDetailBinding view;
    Context context = this;
    ArrayList<DetailModel> arrayList = new ArrayList<>();
    DataModel model;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        view = DataBindingUtil.setContentView(this, R.layout.activity_detail);

        setOnClickListeners();
        showDetail();
    }

    /**
     * This method is used to set click listeners on views
     */
    private void setOnClickListeners() {
        view.ivBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                onBackPressed();
            }
        });
    }

    /***
     * This method is used to get data from the previous activity
     */
    private void showDetail() {
        Intent in = getIntent();
        model = (DataModel) in.getSerializableExtra("model");
        if (model != null) {
            arrayList.add(new DetailModel("Amount:-", model.getAmount()));
            arrayList.add(new DetailModel("Order ID:-", model.getOrderID()));
            arrayList.add(new DetailModel("Transaction Id:-", model.getTransactionId()));
            arrayList.add(new DetailModel("Token:-", model.getToken()));
            arrayList.add(new DetailModel("Transaction by:-", model.getTransactionTitle()));
            arrayList.add(new DetailModel("Email:-", model.getEmail()));
            arrayList.add(new DetailModel("Number:-", model.getNumber()));

            view.recyclerView.setLayoutManager(new LinearLayoutManager(context));
            view.recyclerView.setAdapter(new DetailAdapter(context, arrayList));

        }
    }
}

File: Kotlin/app/src/main/java/com/kotlinpaytaps/DetailActivity.kt
package com.kotlinpaytaps

import android.content.Context
import android.os.Bundle

import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.kotlinpaytaps.adpter.DetailAdapter
import com.kotlinpaytaps.model.DataModel
import com.kotlinpaytaps.model.DetailModel
import kotlinx.android.synthetic.main.activity_detail.*

import java.util.ArrayList

class DetailActivity : AppCompatActivity() {

    internal var context: Context = this
    internal var arrayList = ArrayList<DataModel>()
    internal var model: DetailModel? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)

        init()
    }

    private fun init() {
        getIntentData()
        setOnClickListeners()
    }

    /**
     * This method is used to set click listeners on views
     */
    private fun setOnClickListeners() {
        iv_back.setOnClickListener { v -> onBackPressed() }
    }


    /***
     * This method is used to get data from the previous activity
     */
    private fun getIntentData() {
        val inte = intent
        model = inte.getSerializableExtra("model") as DetailModel
        if (model != null) {
            arrayList.add(DataModel("Amount:-", model!!.amount))
            arrayList.add(DataModel("Order ID:-", model!!.orderID))
            arrayList.add(DataModel("Transaction Id:-", model!!.transactionId))
            arrayList.add(DataModel("Token:-", model!!.token))
            arrayList.add(DataModel("Transaction by:-", model!!.transactionTitle))
            arrayList.add(DataModel("Email:-", model!!.email))
            arrayList.add(DataModel("Number:-", model!!.number))

            recyclerView.setLayoutManager(
                LinearLayoutManager(
                    context,
                    LinearLayoutManager.VERTICAL,
                    false
                )
            )
            // Set the data in adapter
            recyclerView.setAdapter(DetailAdapter(context, arrayList))
        }
    }
}
